### PR TITLE
Parse use intrinsic

### PR DESCRIPTION
**Pull request by TeranIvy**
This PR is for Issue #40. 

I used this opportunity to thoroughly clean `statements.py`.  It now has only one pep8 issue for lambda expression (`src/fparser/statements.py:2429:5: E731 do not assign a lambda expression, use a def`).
I also removed a lot of pylint issues related to more than one  statements per line. This cleanup led to apparent decrease in coverage.

The changes related to bug fix in #40 are in `Use(Statement)` in `statements.py` and `def test_use():` in `test_parser.py`.

### CODE REVIEW

File: src/fparser/statements.py [hunk 1]
```diff
@@ -967,23 +1105,24 @@ def process_item(self):
...
             self.items = split_comma(line, self.item)
         return
 
-    def tofortran(self, isfix=None):
+    def tofortran(self, isfix=None):  # pylint: disable=invalid-name
         tab = self.get_indent_tab(isfix=isfix)
         s = 'USE'
         if self.nature:
-            s += ' ' + self.nature + ' ::'
+            s += ', ' + self.nature + ' ::'
```
Comment (arporter) [inline src/fparser/statements.py]: Found the bug fix! :-) As you've modified this routine please could you add a docstring for it.

File: src/fparser/statements.py [hunk 2]
```diff
@@ -1490,21 +1713,25 @@ def tofortran(self, isfix=None):
...
-    match = re.compile(r'import(\b|\Z)',re.I).match
+    match = re.compile(r'import(\b|\Z)', re.I).match
+
 
 class Forall(Statement):
     """
     FORALL <forall-header> <forall-assignment-stmt>
     <forall-header> = ( <forall-triplet-spec-list> [ , <scalar-mask-expr> ] )
-    <forall-triplet-spec> = <index-name> = <subscript> : <subscript> [ : <stride> ]
+    <forall-triplet-spec> = <index-name>
```
Comment (arporter) [inline src/fparser/statements.py]: I think this should still say `<index-name> = <subscript> : <subscript>....` - perhaps just bring `<index-name>` down to the next line.

File: src/fparser/tests/test_parser.py [hunk 3]
```diff
@@ -334,9 +334,10 @@ def test_nullify():
 
 
 def test_use():
+    ''' Check that we can parse USE statement. '''
     assert parse(Use, 'use a') == 'USE a'
     assert parse(Use, 'use :: a') == 'USE a'
-    assert parse(Use, 'use, intrinsic:: a') == 'USE INTRINSIC :: a'
+    assert parse(Use, 'use, intrinsic:: a') == 'USE, INTRINSIC :: a'
```
Comment (arporter) [inline src/fparser/tests/test_parser.py]: Please add a test for `use, non_intrinsic :: a` and a case where such a use has an `only` clause

Comment (TeranIvy) [reply src/fparser/tests/test_parser.py]: Added.

File: src/fparser/statements.py (hunk 1 above)
Comment (TeranIvy) [reply src/fparser/statements.py]: Thank you, I thought that adding a docstring for the top "class Use(Statement):" would be enough, but now I added them for each method in the class.

**Comment by arporter** [comment]
Hi @TeranIvy, is this one ready for second review now?

File: src/fparser/statements.py (hunk 2 above)
Comment (TeranIvy) [reply src/fparser/statements.py]: Moved.

**Pull request merged**